tests/e2e/util/commonUtils.go

```diff
@@ -0,0 +1,118 @@
+package util
+
+import (
+	"flag"
+	"fmt"
+	"io"
+	"net/http"
+	"os"
+	"os/exec"
+	"path"
+	"path/filepath"
+	"strings"
+
+	"github.com/golang/glog"
+)
+
+const (
+	testSrcDir = "TEST_SRCDIR"
+	pathPrefix = "com_github_istio_istio"
+)
+
+var (
+	manualRun = flag.Bool("manual_run", false, "If runned by bazel run")
+)
+
+// Shell run command on shell and get back output and error if get one
+func Shell(command string) (string, error) {
+	glog.Info(command)
+	parts := strings.Split(command, " ")
+	c := exec.Command(parts[0], parts[1:]...)
+	bytes, err := c.CombinedOutput()
+	if err != nil {
+		glog.V(2).Info(string(bytes))
+		return "", fmt.Errorf("command failed: %q %v", string(bytes), err)
+	}
+	return string(bytes), nil
+}
+
+// HTTPDownload download from src(url) and store into dst(local file)
+func HTTPDownload(dst string, src string) error {
+	glog.Infof("Start downloading from %s to %s ...\n", src, dst)
+	var err error
+	var out *os.File
+	var resp *http.Response
+	out, err = os.Create(dst)
+	if err != nil {
+		return err
+	}
+
+	defer func() {
+		if cerr := out.Close(); err == nil && cerr != nil {
```

Comment:
defer will not change the value that you return. you should just log an error instead.